chalicelib/fs_connection.py

```diff
@@ -42,3 +46,24 @@ def __init__(self, fs_environ, fs_environ_info, test=False):
         else:
             self.ff_s3 = None
             self.ff_keys = None
+
+    def get_object(self, key):
+        """
+        Queries ES for key - checks S3 if it doesn't find it
+        """
+        obj = None
+        if self.connections['es'] is not None:
+            obj = self.connections['es'].get_object(key)
+        if obj is None:
+            obj = self.connections['s3'].get_object(key)
+            if self.connections['es'] is not None:
+                self.connections['es'].put_object(key, obj)
```

Comment:
Feel free to disagree, but I don't think `get_object` should attempt to PUT to ES. An example would be the action record, which seems reasonable to obtain with `FSConnection.get_object`

I get how this provides a backup option to add missing results to ES, but it seems to be a possibly hard-to-debug source of trouble in the future. I think we can say the only things that add items to ES are `RunResult.store_formatted_result` and the migration check

Reply:
Definitely a good point, I see how this could potentially cause problems so I've removed this. My thinking was that in this case the ES connection works like a cache - so in the case that you do request something that isn't in ES chances are you'll want to request it again and it should be moved to ES regardless of the migration schedule.